canonical-web-and-design/maas-ui
# fix(ui): correctly validate size when creating a new partition or logical volume

## Done

- Improved validation for partition/logical volume size
  - Size must be larger than minimum partition size
  - Size must be less than available disk space

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the storage tab of a Ready or Allocated machine
- Add a partition and try to set the size to 1MB
- Check that an error shows what the minimum is
- Try to set the size to more than the available space in the disk
- Check that an error shows what the maximum is
- Do the same thing with a logical volume (you will need to create a volume group first)

## Fixes

Fixes #1987
